### PR TITLE
Feature/tab

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/components/List.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            const keys = Object.keys(groups) as GK[];
+            return keys.sort(groupComparator);
+        },
+        [groups, groupComparator],
+    );
+
+    const children: React.ReactNode[] = [];
+    sortedGroupKeys.forEach((groupKey, i) => {
+        children.push(renderGroup(groupKey, i));
+        children.push(...groups[groupKey].map(renderListItem));
```
Comment (tnagorra) [inline src/components/List.tsx]: While creating children inside grouped list, the key should change when group changes

File: src/components/Tab/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import Button, { ButtonProps } from '../Button';
+import List from '../List';
+
+import styles from './styles.css';
+
+export interface TabProps {
+    children: React.ReactNode;
+}
+
+const tabContext = React.createContext({
```
Comment (tnagorra) [inline src/components/Tab/index.tsx]:
```suggestion
const TabContext = React.createContext({
```

Comment (tnagorra) [reply src/components/Tab/index.tsx]: or we should destructure tabContext before using it in jsx

Bad:
         <tabContext.Provider value={contextValue}> 
Good:
        <Provider value={contexValue}>

File: src/components/List.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    return (
+        <GroupedList
+            {...props}
+        />
+    );
+}
+
+List.defaultProps = {
```
Comment (tnagorra) [inline src/components/List.tsx]: Let's not use defaultProps

File: src/components/Tabs/index.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+import { useThemeClassName } from '../../hooks';
+import styles from './styles.css';
+
+const TabContext = React.createContext({
+    activeTab: undefined,
+    setActiveTab: () => { console.warn('setActiveTab called before it was initialized'); },
+});
+
+type TabKey = string | undefined;
```
Comment (tnagorra) [inline src/components/Tabs/index.tsx]: Shouldn't TabKey always be defined?

**Review by tnagorra — CHANGES_REQUESTED** [review]
Typing issues should be resolved first.

File: src/components/Tabs/index.tsx (hunk 4 above)
Comment (frozenhelium) [reply src/components/Tabs/index.tsx]: No tab can be visible as well

**Review by tnagorra — APPROVED** [review]

**Pull request merged**